src/CodeGen_Posix.cpp

```diff
@@ -115,7 +120,7 @@ CodeGen_Posix::Allocation CodeGen_Posix::create_allocation(const std::string &na
             // We used to do the alloca locally and save and restore the
             // stack pointer, but this makes llvm generate streams of
             // spill/reloads.
-            allocation.ptr = create_alloca_at_entry(i32x8, stack_bytes/32, false, name);
+            allocation.ptr = create_alloca_at_entry(i8, stack_bytes, false, name);
```

Comment:
Per our discussion, I used the actual type, and only re-use allocations when the type matches.